app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,436 @@
+        newMessage.put("id", Integer.toString(m.key.id));
+        newMessage.put("priority", Integer.toString(m.key.priority));
+        newMessage.put("messages", m.message);
+        if (m.actionString != null) {
+            newMessage.put("actions", m.actionString);
+        }
+        a.add(newMessage);
+    }
+
+    /**
+     * Initiates the SnackBar ExpandableListView.
+     */
+    private void buildList() {
+        mList = new ExpandableListView(mContext);
+        mList.setId(R.id.snackbar);
+        setListAppearance();
+        mList.setAdapter(generateAdapter());
+        mTargetParent.addView(mList);
+        hide();
+    }
+
+    /**
+     * Sets the SnackBar ExpandableListView appearance.
+     */
+    private void setListAppearance() {
+        mList.setDivider(null);
+        mList.setChildDivider(null);
+        Drawable icon = mContext.getResources().getDrawable(R.drawable.snackbar_group_indicator);
+        mList.setGroupIndicator(icon);
+        mList.setIndicatorBounds(
+            1200 - Utils.getPixelFromDips(60),
+            1200 - Utils.getPixelFromDips(32)
+        );
+    }
+
+    /**
+     * Instantiates the SnackBarListAdapter with no messages.
+     * @return SnackBarListAdapter
+     */
+    private SnackBarListAdapter generateAdapter() {
+        groups = new ArrayList<>();
+        children = new ArrayList<>();
+        children.add(new ArrayList<HashMap<String, String>>());
+
+        adapter = new SnackBarListAdapter(
+            mContext,
+            groups,                                                   // Creating group List.
+            R.layout.snackbar_group,                                  // Group item layout XML.
+            new String[] {"id", "priority", "messages", "actions"},   // the key of group item.
+            new int[] {                                               // Data under the key goes into this TextView.
+                R.id.snackbar_id,
+                R.id.snackbar_priority,
+                R.id.snackbar_message,
+                R.id.snackbar_action
+            },
+            children,                                                 // childData describes second-level entries.
+            R.layout.snackbar_item,                                   // Layout for sub-level entries(second level).
+            new String[] {"id", "priority", "messages", "actions"},   // Keys in childData maps to display.
+            new int[] {                                               // Data under the keys above go into these TextViews.
+                R.id.snackbar_id,
+                R.id.snackbar_priority,
+                R.id.snackbar_message,
+                R.id.snackbar_action
+            }
+        );
+        return adapter;
+    }
+
+    /**
+     * The SnackBar ExpandableListView Adapter responsible to handling the message data.
+     */
+    private final class SnackBarListAdapter extends SimpleExpandableListAdapter {
```

Comment:
Extending `SimpleExpandableListAdapter` means that you have to do a lot of work fiddling with Hashmaps (as you can see above). I think it would be much cleaner to instead extend `BaseExpandableListAdapter`. It's more boilerplate, and you have to handle the XML inflation yourself, but I think it works much better with the data model that you've got for the messages.